android/src/main/java/com/reactlibrary/JudoReactNativeModule.java

```diff
@@ -2,526 +2,447 @@
+        ReadableMap configurations = options.getMap("configuration");
+        ReadableMap amount = configurations.getMap("amount");
+
+        String amountValue = amount.getString("value");
+        String currencyValue = amount.getString("currency");
+        Currency currency = getCurrency(currencyValue);
+
+        return new Amount.Builder()
+                .setAmount(amountValue)
+                .setCurrency(currency)
+                .build();
     }
 
-    private WritableMap convert(Risks risks) {
-        if (risks == null) {
-            return null;
+    private Currency getCurrency(String value) {
+        switch (value) {
+            case "AED":
+                return Currency.AED;
+            case "AUD":
+                return Currency.AUD;
+            case "BRL":
+                return Currency.BRL;
+            case "CAD":
+                return Currency.CAD;
+            case "CHF":
+                return Currency.CHF;
+            case "DKK":
+                return Currency.DKK;
+            case "EUR":
+                return Currency.EUR;
+            case "HKD":
+                return Currency.HKD;
+            case "HUF":
+                return Currency.HUF;
+            case "JPY":
+                return Currency.JPY;
+            case "NOK":
+                return Currency.NOK;
+            case "NZD":
+                return Currency.NZD;
+            case "PLN":
+                return Currency.PLN;
+            case "SAR":
+                return Currency.SAR;
+            case "SEK":
+                return Currency.SEK;
+            case "SGD":
+                return Currency.SGD;
+            case "QAR":
+                return Currency.QAR;
+            case "USD":
+                return Currency.USD;
+            case "ZAR":
+                return Currency.ZAR;
+            default:
+                return Currency.GBP;
         }
-        WritableMap result = new WritableNativeMap();
-        result.putString("postCodeCheck", risks.getPostCodeCheck());
-        return result;
     }
 
-    private WritableMap convert(OrderDetails orderDetails) {
-        if (orderDetails == null) {
-            return null;
-        }
-        WritableMap result = new WritableNativeMap();
-        result.putString("orderId", orderDetails.getOrderId());
-        result.putString("orderStatus", orderDetails.getOrderStatus().name());
-        result.putString("orderFailureReason", orderDetails.getOrderFailureReason());
-        result.putString("timestamp", orderDetails.getTimestamp());
-        return result;
+    private Reference getReference(ReadableMap options) {
+        ReadableMap configurations = options.getMap("configuration");
+        ReadableMap reference = configurations.getMap("reference");
+
+        String consumerReference = reference.getString("consumerReference");
+        String paymentReference = reference.getString("paymentReference");
+
+        // TODO: MAP METADATA BUNDLE ?
```

Comment:
yup, metadata it's used by KFC to store details about the payment

Reply:
Question is how to map a dynamic value to a Bundle in Android

Reply:
we're expecting Map<String, String> no?

Reply:
In Kotlin:
```
val bundle = bundleOf(
    "KEY_PRICE" to 50.0,
    "KEY_IS_FROZEN" to false
)
```

Reply:
Wasn't metadata freeform ?
I'll change it to a map for now

Reply:
@mpetrenco is metadata freeform in iOS?

Reply:
Changed. Resolve if fixed.